src/core/server/core_usage_stats/core_usage_stats_client.ts

```diff
@@ -0,0 +1,163 @@
+  total: 0,
+  allTypesSelected: Object.freeze({ yes: 0, no: 0 }),
+});
+
+/** @internal */
+export class CoreUsageStatsClient {
+  constructor(
+    private readonly debugLogger: (message: string) => void,
+    private readonly repository: ISavedObjectsRepository
+  ) {}
+
+  public async getUsageStats() {
+    this.debugLogger('getUsageStats() called');
+    let coreUsageStats: CoreUsageStats = {};
+    try {
+      const result = await this.repository.get<CoreUsageStats>(
+        CORE_USAGE_STATS_TYPE,
+        CORE_USAGE_STATS_TYPE
+      );
+      coreUsageStats = result.attributes;
+    } catch (err) {
+      // do nothing
+    }
+    return coreUsageStats;
+  }
+
+  public async incrementSavedObjectsImport({
+    createNewCopies,
+    overwrite,
+  }: IncrementSavedObjectsImportOptions) {
+    const coreUsageStats = await this.getUsageStats();
+    const { apiCalls = {} } = coreUsageStats;
+    const { savedObjectsImport: current = SAVED_OBJECTS_IMPORT_DEFAULT } = apiCalls;
+
+    const attributes = {
+      ...coreUsageStats,
+      apiCalls: {
+        ...apiCalls,
+        savedObjectsImport: {
+          total: current.total + 1,
+          createNewCopiesEnabled: incrementBooleanCounter(
+            current.createNewCopiesEnabled,
+            createNewCopies
+          ),
+          overwriteEnabled: incrementBooleanCounter(current.overwriteEnabled, overwrite),
+        },
+      },
+    };
+    await this.updateUsageStats(attributes);
+  }
+
+  public async incrementSavedObjectsResolveImportErrors({
+    createNewCopies,
+  }: IncrementSavedObjectsResolveImportErrorsOptions) {
+    const coreUsageStats = await this.getUsageStats();
+    const { apiCalls = {} } = coreUsageStats;
+    const {
+      savedObjectsResolveImportErrors: current = SAVED_OBJECTS_RESOLVE_IMPORT_ERRORS_DEFAULT,
+    } = apiCalls;
+
+    const attributes = {
+      ...coreUsageStats,
+      apiCalls: {
+        ...apiCalls,
+        savedObjectsResolveImportErrors: {
+          total: current.total + 1,
+          createNewCopiesEnabled: incrementBooleanCounter(
+            current.createNewCopiesEnabled,
+            createNewCopies
+          ),
+        },
+      },
+    };
+    await this.updateUsageStats(attributes);
+  }
+
+  public async incrementSavedObjectsExport({
+    types,
+    supportedTypes,
+  }: IncrementSavedObjectsExportOptions) {
+    const coreUsageStats = await this.getUsageStats();
+    const { apiCalls = {} } = coreUsageStats;
+    const { savedObjectsExport: current = SAVED_OBJECTS_EXPORT_DEFAULT } = apiCalls;
+    const isAllTypesSelected = !!types && supportedTypes.every((x) => types.includes(x));
+
+    const attributes = {
+      ...coreUsageStats,
+      apiCalls: {
+        ...apiCalls,
+        savedObjectsExport: {
+          total: current.total + 1,
+          allTypesSelected: incrementBooleanCounter(current.allTypesSelected, isAllTypesSelected),
+        },
+      },
+    };
+    await this.updateUsageStats(attributes);
```

Comment:
we should swallow any exceptions from inside the `CoreUsageStatsClient` methods, we don't want these exceptions to bubble up to consuming plugins or for instance the import route causing a 500 because we couldn't persist usage data.

Reply:
Great catch, I'll add it!